src/openloco/companymgr.cpp

```diff
@@ -103,4 +111,78 @@ namespace openloco::companymgr
             }
         }
     }
+
+    // 0x004383ED
+    void updateOwnerStatus()
+    {
+        if (openloco::is_title_mode() || openloco::is_editor_mode())
+        {
+            return;
+        }
+
+        auto company = companymgr::get(_updating_company_id);
+        if (company == nullptr)
+        {
+            return;
+        }
+
+        company->var_12 += 1;
+        if ((company->var_12 & 0x7F) != 0)
+            return;
+
+        for (size_t i = 0; i < WindowManager::count(); i++)
+        {
+            auto w = WindowManager::get(i);
+
+            if (w->type != WindowType::vehicle)
+                continue;
+
+            auto vehicle = thingmgr::get<openloco::vehicle>(w->number);
+            if (vehicle->x == -0x8000)
+                continue;
+
+            if (vehicle->owner != _updating_company_id)
+                continue;
+
+            game_commands::do_73(vehicle->id);
+            return;
+        }
+
+        auto main = WindowManager::getMainWindow();
+        if (main == nullptr)
+            return;
+
+        auto viewport = main->viewports[0];
+        if (viewport == nullptr)
+            return;
+
+        int16_t posX = viewport->x + viewport->width / 2;
+        int16_t posY = viewport->y + viewport->height / 2;
+
+        registers r1;
+        r1.ax = posX;
+        r1.bx = posY;
+        call(0x0045F1A7, r1);
+        ui::viewport* vp = (ui::viewport*)r1.edi;
+
+        if (posX != -0x8000 && viewport == vp)
+        {
+            // The result of this function appears to be ignored
+            map::tile_element_height(posX, posY);
+        }
+        else
+        {
+            registers r2;
+
+            r2.ax = viewport->view_x + viewport->view_width / 2;
+            r2.bx = viewport->view_y + viewport->view_height / 2;
+            r2.edx = gCurrentRotation;
```

Comment:
For reference, the call I'd like replaced with an accessor/getter call, as mentioned earlier.